File: python/1090.py
n = int(input())
pos = [list(map(int, input().split())) for _ in range(n)]
ans = [1e9] * n
for xs in pos:
    for ys in pos:
        distance = []
        for x,y in pos:
            distance.append(abs(xs[0] - x) + abs(ys[1] - y))
        distance.sort()
        tmp = 0
        for i in range(n):
            tmp += distance[i]
            ans[i] = min(ans[i],tmp)
print(*ans)

File: js/Baekjoon/1090.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
const N = parseInt(input[0]);
const pos = input.slice(1).map((e) => e.split(" ").map(Number));

let ans = new Array(N).fill(10e9);
for (let i = 0; i < N; i++) {
  for (let j = 0; j < N; j++) {
    let distance = [];
    for (let k = 0; k < N; k++) {
      distance.push(
        Number(
          Math.abs(pos[i][0] - pos[k][0]) + Math.abs(pos[j][1] - pos[k][1])
        )
      );
    }
    distance.sort((a, b) => a - b);
    let tmp = 0;
    for (let k = 0; k < N; k++) {
      tmp += distance[k];
      ans[k] = Math.min(ans[k], tmp);
    }
  }
}
console.log(ans.join(" "));
